### PR TITLE
style(DatePicker): update to latest design

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-2608]

[Design](https://www.figma.com/file/5SCTOPrCDcHuk5We091GBn/Product-Library)

### Description

Update design of DatePicker to latest

### How to test

- Compare [design](https://www.figma.com/file/5SCTOPrCDcHuk5We091GBn/Product-Library) with the [storybook](https://picasso.toptal.net/FX-2608-date-picker/?path=/story/components-datepicker--datepicker)

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | image |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Annotate all `props` in component with documentation
- Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2608]: https://toptal-core.atlassian.net/browse/FX-2608?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

File: packages/picasso/src/Calendar/styles.ts [hunk 1]
```diff
@@ -27,9 +30,11 @@ export default ({ palette }: Theme) =>
     weekDays: {
       display: 'flex',
       textAlign: 'center',
-      fontSize: '0.75em',
+      fontSize: '0.75rem',
+      lineHeight: '1.5',
       textTransform: 'uppercase',
       color: palette.grey.main2,
+      paddingTop: rem('3px'),
```
Comment (dmaklygin) [inline packages/picasso/src/Calendar/styles.ts:37]: three different measurements here: px, rem, and scale.
I would like to have a more consistent solution...

**Review by dmaklygin — APPROVED** [review]

**Comment by OndrejTuma** [comment]
I see that [in design](https://www.figma.com/file/5SCTOPrCDcHuk5We091GBn/Product-Library?node-id=1927%3A15035) there are round borders in selection 🤔 
image

**Comment by TomasSlama** [comment]
@vshyrokov checked in SP ❤️

**Comment by TomasSlama** [comment]
@OndrejTuma great catch, totally missed that one. ✅ fixed.

**Review by OndrejTuma — APPROVED** [review]
Nice work 👍

File: packages/picasso/src/Calendar/styles.ts [hunk 2]
```diff
@@ -22,14 +25,17 @@ export default ({ palette }: Theme) =>
...
       border: 'none',
-      outline: 0
+      outline: 0,
+      borderRadius: sizes.borderRadius.small
     },
     weekDays: {
       display: 'flex',
       textAlign: 'center',
-      fontSize: '0.75em',
+      fontSize: rem('12px'),
```
Comment (LashaJini) [inline packages/picasso/src/Calendar/styles.ts:34]: Didn't we agree to use `em`s as much as possible :thinking:

**Review by LashaJini — APPROVED** [review]

File: packages/picasso/src/Calendar/styles.ts (hunk 2 above)
Comment (TomasSlama) [reply packages/picasso/src/Calendar/styles.ts:34]: I don't think so, I think we should you `rem`

**Review by ozgurkececioglu — APPROVED** [review]

**Pull request merged**